app/src/app/services/city.service.ts

```diff
@@ -0,0 +1,92 @@
+import { Injectable } from '@angular/core';
+import {
+  HttpClient,
+  HttpErrorResponse,
+  HttpHeaders,
+  HttpResponse,
+} from '@angular/common/http';
+import { LoggerService } from './logger.service';
+import { catchError, Observable, throwError } from 'rxjs';
+import { environment } from 'src/environments/environment';
+import { Cities } from '../types/cities';
+import { City } from '../types/city';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class CityService {
+  constructor(
+    private logger: LoggerService,
+    private http: HttpClient,
+    private authService: AuthService
+  ) {}
+
+  getAllCities(pageNumber: number, pageSize: number): Observable<Cities> {
+    const headers = new HttpHeaders()
+      .set('Authorization', `Bearer ${this.authService.userProfile.value.jwt}`)
+      .set('Content-Type', 'application/json');
+
+    const params = {
+      pageNumber: pageNumber,
+      pageSize: pageSize,
+    };
+
+    return this.http
+      .get<Cities>(`${environment.apiUrl}/Cities`, {
+        headers: headers,
+        params: params,
+      })
+      .pipe(
+        catchError((error: HttpErrorResponse) => {
+          this.logger.error(error.message);
+          return throwError(() => error);
+        })
+      );
+  }
+
+  searchCities(
+    pageNumber: number,
+    pageSize: number,
+    name: string
+  ): Observable<Cities> {
+    const headers = new HttpHeaders()
```

Comment:
Just a note here - using an interceptor would probably reduce a little bit of duplication here (see https://angular.io/guide/http-interceptor-use-cases).